### ISSUE TITLE
[Bug]: datacoord crash with panic: runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#17823]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: 2.1.0-20220625-4e2b2bfa
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.1.0.dev83
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: 

config.common.retentionDuration: 40
```
```


### Current Behavior

1. Run all compaction test cases with 4 processes
```
pytest testcases/test_compaction.py -n 4 --host=10.98.0.x
```
2. Test result:
```
[2022-06-27 12:52:27 - ERROR - pymilvus.client.grpc_handler]: error_code: UnexpectedError
reason: "CreateCollection failed: err: find no available datacoord, check datacoord state\n, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace\n/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:259 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall\n/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:458 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).WatchChannels\n/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:728 github.com/milvus-io/milvus/internal/rootcoord.(*Core).SetDataCoord.func4\n/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:269 github.com/milvus-io/milvus/internal/rootcoord.(*CreateCollectionReqTask).Execute\n/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:63 github.com/milvus-io/milvus/internal/rootcoord.executeTask.func1\n/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit\n"
 (grpc_handler.py:192)
[2022-06-27 12:52:27 - ERROR - pymilvus.decorators]: RPC error: [create_collection], <MilvusException: (code=1, message=CreateCollection failed: err: find no available datacoord, check datacoord state
, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:259 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall
/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:458 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).WatchChannels
/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:728 github.com/milvus-io/milvus/internal/rootcoord.(*Core).SetDataCoord.func4
/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:269 github.com/milvus-io/milvus/internal/rootcoord.(*CreateCollectionReqTask).Execute
/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:63 github.com/milvus-io/milvus/internal/rootcoord.executeTask.func1
/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit
)>, <Time:{'RPC start': '2022-06-27 12:52:27.604624', 'RPC error': '2022-06-27 12:52:27.745441'}> (decorators.py:78)
[2022-06-27 12:52:27 - ERROR - ci_test]: Traceback (most recent call last):
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 22, in inner_wrapper
    res = func(*args, **kwargs)
  File "/Users/nausicca/projects/milvus/tests/python_client/utils/api_request.py", line 56, in api_request
    return func(*arg, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/orm/collection.py", line 138, in __init__
    conn.create_collection(self._name, fields=schema.to_dict(), shards_num=self._shards_num,
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 61, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 44, in handler
    return func(self, *args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 79, in handler
    raise e
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/decorators.py", line 75, in handler
    return func(*args, **kwargs)
  File "/Users/nausicca/.virtualenvs/milvus/lib/python3.8/site-packages/pymilvus/client/grpc_handler.py", line 193, in create_collection
    raise MilvusException(status.error_code, status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=CreateCollection failed: err: find no available datacoord, check datacoord state
, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
/go/src/github.com/milvus-io/milvus/internal/util/grpcclient/client.go:259 github.com/milvus-io/milvus/internal/util/grpcclient.(*ClientBase).ReCall
/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/client/client.go:458 github.com/milvus-io/milvus/internal/distributed/datacoord/client.(*Client).WatchChannels
/go/src/github.com/milvus-io/milvus/internal/rootcoord/root_coord.go:728 github.com/milvus-io/milvus/internal/rootcoord.(*Core).SetDataCoord.func4
/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:269 github.com/milvus-io/milvus/internal/rootcoord.(*CreateCollectionReqTask).Execute
/go/src/github.com/milvus-io/milvus/internal/rootcoord/task.go:63 github.com/milvus-io/milvus/internal/rootcoord.executeTask.func1
/usr/local/go/src/runtime/asm_amd64.s:1371 runtime.goexit
)>
 (api_request.py:35)
[2022-06-27 12:52:27 - ERROR - ci_test]: (api_response) : <MilvusException: (code=1, message=CreateCollection failed: err: find no available datacoord, check datacoord state
, /go/src/github.com/milvus-io/milvus/internal/util/trace/stack_trace.go:51 github.com/milvus-io/milvus/internal/util/trace.StackTrace
/go/src/github.com/milvus-io/milvus/internal/util...... (api_request.py:36)

 testcases/test_compaction.py::TestCompactionOperation.test_compact_and_insert ⨯ 
```
3. Datacoord crash 
```
[2022/06/27 04:51:38.964 +00:00] [ERROR] [services.go:807] ["failed to complete compaction"] [planID=434190547473661955] [error="plan 434190547473661955 is not found"] [stack="github.com/milvus-io/milvus/internal/datacoord.(*Server).CompleteCompaction\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/services.go:807\ngithub.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).CompleteCompaction\n\t/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:296\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataCoord_CompleteCompaction_Handler.func1\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5133\ngithub.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38\ngithub.com/milvus-io/milvus/internal/proto/datapb._DataCoord_CompleteCompaction_Handler\n\t/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5135\ngoogle.golang.org/grpc.(*Server).processUnaryRPC\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1282\ngoogle.golang.org/grpc.(*Server).handleStream\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1616\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2\n\t/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:921"]
[2022/06/27 04:51:38.964 +00:00] [INFO] [services.go:434] ["receive DropVirtualChannel request"] ["channel name"=by-dev-rootcoord-dml_13_434190535099154433v0]
[2022/06/27 04:51:38.964 +00:00] [INFO] [services.go:434] ["receive DropVirtualChannel request"] ["channel name"=by-dev-rootcoord-dml_21_434190545691607041v0]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c4563c]

goroutine 167 [running]:
github.com/milvus-io/milvus/internal/datacoord.isSegmentHealthy(...)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/meta.go:1079
github.com/milvus-io/milvus/internal/datacoord.(*meta).mergeDropSegment(0xc0004f36c0, 0xc000e2e9b0, 0xc000add110)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/meta.go:466 +0x9dc
github.com/milvus-io/milvus/internal/datacoord.(*meta).UpdateDropChannelSegmentInfo(0xc0004f36c0, 0xc000f160c0, 0x2c, 0xc000c26968, 0x1, 0x1, 0x0, 0x0)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/meta.go:426 +0x19c
github.com/milvus-io/milvus/internal/datacoord.(*Server).DropVirtualChannel(0xc0000ecd00, 0x436a198, 0xc00037eed0, 0xc00032a960, 0x3f76420, 0x0, 0x439e510)
	/go/src/github.com/milvus-io/milvus/internal/datacoord/services.go:463 +0x6c6
github.com/milvus-io/milvus/internal/distributed/datacoord.(*Server).DropVirtualChannel(0xc000aff980, 0x436a198, 0xc00037eed0, 0xc00032a960, 0xc000aff980, 0x4332fa0, 0xc00032b0e0)
	/go/src/github.com/milvus-io/milvus/internal/distributed/datacoord/service.go:326 +0x52
github.com/milvus-io/milvus/internal/proto/datapb._DataCoord_DropVirtualChannel_Handler.func1(0x436a198, 0xc00037eed0, 0x3e746c0, 0xc00032a960, 0x3fd2fd8, 0xd, 0x4031343, 0x2f)
	/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5241 +0x89
github.com/grpc-ecosystem/go-grpc-middleware/tracing/opentracing.UnaryServerInterceptor.func1(0x436a198, 0xc00037ea50, 0x3e746c0, 0xc00032a960, 0xc0003405c0, 0xc00037cb10, 0xc000f2bba0, 0x9acea6, 0x3e260c0, 0xc00037ea50)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/tracing/opentracing/server_interceptors.go:38 +0x119
github.com/milvus-io/milvus/internal/proto/datapb._DataCoord_DropVirtualChannel_Handler(0x3f76420, 0xc000aff980, 0x436a198, 0xc00037ea50, 0xc00033cd20, 0xc000b065d0, 0x436a198, 0xc00037ea50, 0xc000f14240, 0xb8)
	/go/src/github.com/milvus-io/milvus/internal/proto/datapb/data_coord.pb.go:5243 +0x150
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000299180, 0x4383758, 0xc000632ea0, 0xc000f0c7e0, 0xc000b00930, 0x593e6d8, 0x0, 0x0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1282 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc000299180, 0x4383758, 0xc000632ea0, 0xc000f0c7e0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:1616 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000af8410, 0xc000299180, 0x4383758, 0xc000632ea0, 0xc000f0c7e0)
	/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:921 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.44.0/server.go:919 +0x1fd
```

**Datacoord log:**  
[datacoord.log](https://github.com/milvus-io/milvus/files/8988787/datacoord.log)


### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

**Comment by xiaofan-luan** [comment]
/assign @XuanYang-cn 
Could you help on verify this

**Comment by ThreadDao** [comment]
Fixed `2.1.0-20220628-2e637388`
